refactor: move data engine to redux

Mohammer5 opened the pull request:
I've removed the `connected-react-router` dep completely as it was entirely not needed (manual pushing can be done through the `history` instance with `withRouter`).

I also moved the data engine stuff from components to redux thunks, which allows for loading the schemas all at once in the beginning and then let all other components use that (I've used the new `react-redux` hooks, which are quite nice btw)

HendrikThePendric commented:
Oh yeah, and regarding the removal of `connected-react-router`: I am on board with that too. 

I quite liked having some router state available in the actions (via `getState()`) when we were working on the `reports-app`. However, in practice quite a lot of router-related stuff (i.e. `match.params`) was not available via the router state and we ended up having to fly through some weird hoops to get stuff to work,[like we did here](https://github.com/dhis2/reports-app/blob/master/src/redux/selectors/standardReport/getEditFormInitialValues.js#L12-L17).

So retrospectively it would have probably made more sense to a) not have used the connected router, and b) just pass route-related action-arguments from the component. So not having this in the maintenance app makes a lot of sense to me.

Mohammer5 commented:
@HendrikThePendric I've cleaned up a bit, fixed a permissions check error in the `NavigationLink` component and added permission checks to `All` and `Overview`.
That means I've added changes that have not been yet approved, could you please have a look if this is okay to you as well?
